util/util.go

```diff
@@ -0,0 +1,195 @@
+		privKeyPEM := pem.EncodeToMemory(&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: privKeyBytes,
+		})
+		err = ioutil.WriteFile(path, privKeyPEM, 0600)
+		if err != nil {
+			return nil, err
+		}
+		log.Println("New private key generated and saved to", path)
+
+		libp2pPrivKey, _, err := crypto.KeyPairFromStdKey(privKey);
+		if(err != nil){
+			return nil, err
+		}
+
+		return libp2pPrivKey, nil;
+	} else if err != nil {
+		// Some other error occurred when trying to read the file
+		return nil, err
+	}
+
+	// Private key file exists, let's read it
+	privKeyBytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	block, _ := pem.Decode(privKeyBytes)
+	if block == nil || block.Type != "RSA PRIVATE KEY" {
+		log.Println("Private key file is of invalid format")
+		return nil, errors.New("private key file is of invalid format")
+	}
+	privKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	log.Println("Existing private key loaded from", path)
+
+	libp2pPrivKey, _, err := crypto.KeyPairFromStdKey(privKey);
+	if(err != nil){
+		return nil, err
+	}
+
+	return libp2pPrivKey, nil;
+}
+
+/*
+ * Check for peers who have announced themselves on the DHT. 
+ * If the DHT is running in server mode, then we will announce ourselves and check for 
+ * others who have announced as well.
+ *
+ * Parameters:
+ *   context: The context
+ *   h: libp2p host
+ *   kDHT: the libp2p ipfs DHT object to use
+ *   advertise: the string to use to check for others who have announced themselves. If
+ * 				DHT is in server mode then that string will be used to announce ourselves as well.
+ *
+ */
+func DiscoverPeers(ctx context.Context, h host.Host, kDHT *dht.IpfsDHT, advertise string) {
+	routingDiscovery := drouting.NewRoutingDiscovery(kDHT);
+	if(kDHT.Mode() == dht.ModeServer){
+		dutil.Advertise(ctx, routingDiscovery, advertise);
+	}
+
+	// Look for others who have announced and attempt to connect to them
+	for {
+		fmt.Println("Searching for peers...")
+		peerChan, err := routingDiscovery.FindPeers(ctx, advertise)
+		if err != nil {
+			panic(err)
+		}
+		for peer := range peerChan {
+			if peer.ID == h.ID() {
+				continue // No self connection
+			}
+			err := h.Connect(ctx, peer)
+			if err != nil {
+				fmt.Printf("Failed connecting to %s, error: %s\n", peer.ID, err)
+			} else {
+				fmt.Println("Connected to:", peer.ID)
+			}
+		}
+		time.Sleep(time.Second * 10)
+	}
+}
+
+
+/*
+ * Searches for file bootstrap.peer and parses it to get multiaddrs of bootstrap peers.
+ * Each line is a mutliaddr.
+ *
+ * Returns:
+ *   A slice of libp2p multiaddrs
+ * Author: Erick
+ */
+func ReadBootstrapPeers() []multiaddr.Multiaddr {
+	peers := []multiaddr.Multiaddr{}
+
+	file, err := os.Open("bootstrap.peers")
+	if err != nil {
+		panic(err)
+	}
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
+
+	for scanner.Scan() {
+		// might have to strip the newline char ?
```

Comment:
This looks like a TODO. Remove the comment if it is not required/already done. Otherwise, add "TODO"